station-world-events-v0/src/main/java/net/modificationstation/stationapi/api/celestial/CelestialEvent.java

```diff
@@ -0,0 +1,270 @@
+package net.modificationstation.stationapi.api.celestial;
+
+import lombok.Getter;
+import net.minecraft.world.World;
+
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Random;
+
+/**
+ * Contains the most important logic for automatically starting and stopping the event.
+ * It is recommended to add the event to CelestialTimeManger for automatic management.
+ * Inheritance is possible to add custom logic.
+ */
+public class CelestialEvent {
+    @Getter
+    private final String name;
+    private final int frequency;
+    private float chance = 1;
+    private int dayLength = 24000;
+    private int dayOffset = 0;
+    private int startingDaytime = 0;
+    private int endingDaytime = 0;
+    private int extraDays = 0;
+    @Getter
+    private boolean active;
+    private boolean initializationNeeded = true;
+    private final List<CelestialEvent> incompatibleEvents = new LinkedList<>();
+    private final List<CelestialEvent> dependencies = new LinkedList<>();
+    public final World world;
```

Comment:
Getting a feeling this can be implemented better. Is there really a need to re-instantiate events each time the world is changed?

Maybe making the events long-lived objects and managing them through a CelestialEventRegistry would make more sense.

Reply:
One idea I have is to only reload world-specific variables using a custom method in the event